src/transport/tcp/tcp_transport.cpp

```diff
@@ -17,32 +17,39 @@ namespace libp2p::transport {
     }
 
     auto conn = std::make_shared<TcpConnection>(*context_);
-    auto rendpoint = detail::makeEndpoint(address);
-    if (!rendpoint) {
-      return handler(rendpoint.error());
-    }
 
-    conn->resolve(rendpoint.value(),
-                  [self{shared_from_this()}, conn, handler{std::move(handler)},
-                   remoteId](auto ec, auto r) mutable {
-                    if (ec) {
-                      return handler(ec);
-                    }
-
-                    conn->connect(
-                        r,
-                        [self, conn, handler{std::move(handler)}, remoteId](
-                            auto ec, auto &e) mutable {
-                          if (ec) {
-                            return handler(ec);
-                          }
-
-                          auto session = std::make_shared<UpgraderSession>(
-                              self->upgrader_, std::move(conn), handler);
-
-                          session->secureOutbound(remoteId);
-                        });
-                  });
+    auto [host, port] = detail::getHostAndPort(address);
+
+    auto connect = [self{shared_from_this()}, conn, handler{std::move(handler)},
+                    remoteId](auto ec, auto r) mutable {
+      if (ec) {
+        return handler(ec);
+      }
+
+      conn->connect(r,
+                    [self, conn, handler{std::move(handler)}, remoteId](
+                        auto ec, auto &e) mutable {
+                      if (ec) {
+                        return handler(ec);
+                      }
+
+                      auto session = std::make_shared<UpgraderSession>(
+                          self->upgrader_, std::move(conn), handler);
+
+                      session->secureOutbound(remoteId);
```

Comment:
It returns handler(ec) in case ec is true, but what does it return when it's not?

Reply:
handler is void function. Here we should not return anything